### ISSUE TITLE
Reference: Priorities list (to be updated every week by Susan and Mihaela)

### ISSUE THREAD

**Issue by ilovan** [cwrc/CWRC-Writer#252]
Thursday Oct 16 priorities list  (please consult milestones as well)
1. schema integration ( #240 Schema lookup fails on non-defined elements)
2. #249 highlighting of items in editing pane doesn't always correspond to the tag selected in structure panel
3. #248 remove tag (right click or control click) when applied to an XML-RDF entity removes the contents as well as the tag
4. #247 Change tag (right click or ctrl click) doesn't work for XML-RDF entities
5. #245 Both TEST and DEV: Note placeholder not displaying properly in neither Firefox nor Chrome - 
6. #244 Both TEST and DEV: Overlap mode broken
7. #251 Grey out the CWRC Writer interface while different components of the editor are still loading upon launch
8. #242 CWRC Writer - Islandora authentication issue

**Comment by ajmacdonald** [comment]
@ilovan @SusanBrown I believe most of these issues are either closed or need confirmation as complete. Is there a consensus on the current priorities?

**Comment by susanirenebrown** [comment]
I believe Mihaela is working on something for a standup with James soon. 

Mihaela will probably be going through these today and coming up with another draft list of priorities, I would think.

Stay tuned.

**Comment by ilovan** [comment]
Wednesday, Jan 21 (remaining development hours: 58.5)

[Note to @ajmacdonald : the list might change over the next few days, as we are only now starting to test the CWRC entry schema integration]

Milestone Jan 22- 28:

1. #244 Both TEST and DEV: Overlap mode broken 
2. #286 Overlap mode defaulting back to no overlap if no overlapping entities are added
3. #282 CWRC ENTRY documents: no css  
4. #283 Orlando: when tags are visible, Ps inside SHORTPROSE invisible

Pending additional instructions from CWRC:
5. #264 Pop-up warning when copying and pasting in CWRC Writer (content of instructions, links to relevant documentation sections)
6. #245 Both TEST and DEV: Note placeholder not displaying properly (Firefox, Chrome)

**Comment by ilovan** [comment]
Thursday, Jan 29 (remaining development hours: 51.5)

Milestone Jan 29- Feb 5:

[Note to @ajmacdonald : During this milestone, additional work may come up as a result of Jeff's changes to the entity dialogues; if or when such work comes up, it will take precedence over the issues listed below.]

1. #286 Overlap mode defaulting back to no overlap if no overlapping entities are added
2. #265 Make the markup panel scroll synchronously with the text area


Pending additional instructions from CWRC:
3. #264 Pop-up warning when copying and pasting in CWRC Writer (content of instructions, links to relevant documentation sections)
4. #245 Both TEST and DEV: Note placeholder not displaying properly (Firefox, Chrome)

**Comment by ilovan** [comment]
Monday, March 2 (remaining hours: 47)

CWRC Dialogs issues
1. cwrc/CWRC-Dialogs#59 : "edit" & "select" buttons should not be clickable unless an item is selected in the search result listing
2. cwrc/CWRC-Dialogs#58 : Review code for hard-coded URLs and place logic into delegator
3. cwrc/CWRC-Dialogs#60: CWRC-Writer update: pass along following information

Pending additional instructions from CWRC:
4. #289 Update CEWW template template from: http://cwrc.ca/templates/ceww_new_entry_template.xml
5. #278 Push DEV to TEST

**Comment by jchartrand** [comment]
Hi Mihaela,

Are these issues for Andrew to look at, even though they’re classified as CWRCDialog issues?  If some of the changes (or parts of the changes) are for the CWRCWriter, could we create new issues in the CWRCWriter repo to tease out what's specific to the CWRCWriter, and to keep track of CWRCWriter changes in the CWRCWriter repo?

james

**Comment by ilovan** [comment]
Tuesday, March 2 (remaining hours: 47)

Milestone March 3-10, 2015

CWRC Dialogs issues
1. #290 : "edit" & "select" buttons should not be clickable unless an item is selected in the search result listing
2. #291 : Review code for hard-coded URLs and place logic into delegator
3. #292 : CWRC-Writer update: pass along following information

Pending additional instructions from CWRC:
4. #289 Update CEWW template template from: http://cwrc.ca/templates/ceww_new_entry_template.xml
5. #293 Push DEV to TEST

**Comment by ilovan** [comment]
Wednesday, March 4  (remaining hours: 47)

Milestone March 3-10, 2015

CWRC Dialogs issues
1. #294 Add licensing information to the CWRC Writer repository
2. #290 : "edit" & "select" buttons should not be clickable unless an item is selected in the search result listing
3. #291 : Review code for hard-coded URLs and place logic into delegator
4. #292 : CWRC-Writer update: pass along following information

Pending additional instructions from CWRC:
5. #289 Update CEWW template template from: http://cwrc.ca/templates/ceww_new_entry_template.xml
6 #293 Push DEV to TEST

**Comment by ilovan** [comment]
Thursday, March 12 (remaining hours: 43)

Milestone March 12-19, 2015

CWRC Dialogs issues
1. #290 : "edit" & "select" buttons should not be clickable unless an item is selected in the search result listing
2. #291 : Review code for hard-coded URLs and place logic into delegator
3. #289 Update CEWW template template from: http://cwrc.ca/templates/ceww_new_entry_template.xml
4. #293 Push DEV to TEST

**Comment by ilovan** [comment]
Thursday, March 26 (remaining hours: 37?)
Milestone: March 27- April 3, 2015

[CWRC-Writer Islandora integration pull requests from dgi have priority]

1. #289 Update CEWW template template 
2. #297 Embeded Dialogs: Country not dispalying. bug
3. #296 Top of the cascading rightclick menu is cut off (search box, first tags - "hidden") bug
4. #299 Changing schema for current document is broken

Pending further discussion with CWRC:
5. #303  CWRC entry: Annotate person entity on a Birthname tag high priority question
6. #301 Avoiding tag redundancy when adding annotations to existing tag enhancement high priority

After #303 and #301 are dealt with:
7. #293 Push DEV to TEST

**Comment by ilovan** [comment]
Tuesday, Jul 21:
Milestone: July 20 - 27, 2015

1. #336 Read-only version: remove mode notification
2. #337 Read-only version:  remove warning notification when trying to close the URL
3.  #338 Read-only and annotation versions: keep the structure panel available, but collapsed (deselected for the annotation version) by default
4. #333 Editor mode: the way in which the mode info is stored with the object or managed should be reviewed
5. #329 Default mode should be XML and RDF (no overlap)
6. #320 Reconsider help popup text
7. #339 All versions: make look compatible with the rest of the redesigned CWRC interface
8. #335 Settings: Make Lato font family available in the font dropdown.

**Comment by SusanBrown** [comment]
1. #340 Upgrade to latest Tiny MCE (Andrew)
2. #301 Avoiding tag redundancy: please respond (may be James)
3. #325 config question: please respond (may be Andrew or James)
4. #345 Entities don't have Attendant URIs (Andrew)
5. #348 Entities don't have URIs in Orlando docs (Andrew)
6. #343 rationalize top menu (guidance coming from Susan; skip if it hasn't arrived)
7. #338 Read-only and annotation versions: keep the structure panel available, but collapsed (deselected for the annotation version) by default (Andrew)
8. #339 All versions: make look compatible with the rest of the redesigned CWRC interface
9. #335 Settings: Make Lato font family available in the font dropdown.
10. #347 Start administrative documentation with config noted in #333 (James)

**Comment by ilovan** [comment]
Tuesday, Aug 4, 2015:

1. #340 Upgrade to latest Tiny MCE (Andrew)
2. #301 Avoiding tag redundancy: please respond (Andrew to reply to the latest post)
3. #345 Entities don't have Attendant URIs (Andrew)
  * #348 Entities don't have URIs in Orlando docs (Andrew) - may be same as #345  (MVI- please confirm if this has been tested while I was away)
4. #338 Read-only and annotation versions: keep the structure panel available, but collapsed (deselected for the annotation version) by default (Andrew)
5. #339 All versions: make look compatible with the rest of the redesigned CWRC interface
6. #335 Settings: Make Lato font family available in the font dropdown.
7. #347 Start administrative documentation with config noted in #333 (James)

Blockers:
8. #343 rationalize top menu (input from CWRC to arrive by early next week)

**Comment by ilovan** [comment]
Friday, Aug 7, 2015:

1. #352 Make images displayable in CWRC-Writer?
2. #353 Does CWRC_Writer support foldout styling?
3. #340 Upgrade to latest Tiny MCE (Andrew)
4. #301 Avoiding tag redundancy: please respond (Andrew to reply to the latest post)
5. #345 Entities don't have Attendant URIs (Andrew)
 *  #348 Entities don't have URIs in Orlando docs (Andrew) - may be same as #345 (MVI- please confirm if this has been tested while I was away)
6. #338 Read-only and annotation versions: keep the structure panel available, but collapsed (deselected for the annotation version) by default (Andrew)
7. #339 All versions: make look compatible with the rest of the redesigned CWRC interface
8. #347 Start administrative documentation with config noted in #333 (James)

Blockers:
9. #343 rationalize top menu (input from CWRC to arrive by early next week)

**Comment by ilovan** [comment]
Monday, Aug 17, 2015:

1. #301 Avoiding tag redundancy
2. #338 Read-only and annotation versions: keep the structure panel available, but collapsed (deselected for the annotation version) by default
3. #347 Start administrative documentation with config noted in #333
4. #358 All versions (read-only, annotation, editor): Add full screen mode.
5. #356 Annotator and read-only versions: checking the "show tags" option in settings trigers the Editor mode warning message
6. #357 CWRC entry schema: orgtype attribute not available for ORGNAME

CWRC:
7. #352 Make images displayable in CWRC-Writer (mappings to be provided by CWRC)
8. #343 rationalize top menu (input from CWRC pending)

**Comment by ilovan** [comment]
Tuesday, Aug 25, 2015:

1.  #301 Avoiding tag redundancy
2. #338 Read-only and annotation versions: keep the structure panel available, but collapsed (deselected for the annotation version) by default
3. #352 Make images displayable in CWRC-Writer (mappings to be provided by CWRC)
4. #347 Start administrative documentation with config noted in #333 
5. #359 Update to TinyMCE caused multiple interface issues to the Integrated version
6. #211 Role, certainty, etc. pop-up needs to resize to fit the current browser window.
7. #355 Read-only mode: selecting text allows you to alter doc text

CWRC:
8.  #343 rationalize top menu (input from CWRC pending)
9. #359 Update to TinyMCE caused multiple interface issues to the Integrated version

**Comment by ilovan** [comment]
Monday, Aug 31, 2015:

  1.  #301 Avoiding tag redundancy
  2.  #338 Read-only and annotation versions: keep the structure panel available, but collapsed (deselected for the annotation version) by default
  3.  #352 Make images displayable in CWRC-Writer (mappings to be provided by CWRC)
  4.  #347 Start administrative documentation with config noted in #333
  5.  #211 Role, certainty, etc. pop-up needs to resize to fit the current browser window.
  6.  #355 Read-only mode: selecting text allows you to alter doc text

CWRC:
7. #359 Update to TinyMCE caused multiple interface issues to the Integrated version 
8. #343 rationalize top menu (input from CWRC pending)

**Comment by ilovan** [comment]
Tuesday, Sep 8, 2015:

1. #301 Avoiding tag redundancy
2. #338 Read-only and annotation versions: keep the structure panel available, but collapsed (deselected for the annotation version) by default
3. #347 Start administrative documentation with config noted in #333
4. #211 Role, certainty, etc. pop-up needs to resize to fit the current browser window.
5. #355 Read-only mode: selecting text allows you to alter doc text
6. #360 Chrome: clicking on a tag in the structure panel doesn't turn it bold
7. #361 Pasted tags not showing up in the structure panel
8. #264 Pop-up warning message when copying/pasting
9. #171 Add copy tag and contents to the right-click options

CWRC:
10. #343 rationalize top menu (input from CWRC pending)

**Comment by ilovan** [comment]
Tuesday, Sep 8, 2015 (revised):

1. #301 Avoiding tag redundancy
2. #338 Read-only and annotation versions: keep the structure panel available, but collapsed (deselected for the annotation version) by default
3. #347 Start administrative documentation with config noted in #333
4. #211 Role, certainty, etc. pop-up needs to resize to fit the current browser window
5. #360 Chrome: clicking on a tag in the structure panel doesn't turn it bold
6. #361 Pasted tags not showing up in the structure panel
7. #264 Pop-up warning message when copying/pasting
8. #171 Add copy tag and contents to the right-click options

CWRC:
9. #343 rationalize top menu (input from CWRC pending)

**Comment by ilovan** [comment]
Monday, Sep 21,  2015:

Maximum priority (dgi dependencies):
1. #362 Save and Exit icon and REST Endpoints

Regular:
2. #338 Read-only and annotation versions: keep the structure panel available, but collapsed (deselected for the annotation version) by default
3. #360 Chrome: clicking on a tag in the structure panel doesn't turn it bold
4. #361 Pasted tags not showing up in the structure panel
5. #264 Pop-up warning message when copying/pasting
6. #171 Add copy tag and contents to the right-click options

CWRC:
7. #301 Avoiding tag redundancy
8. #343 rationalize top menu (input from CWRC pending)

**Comment by ilovan** [comment]
Tuesday, Sep 29, 2015:

remaining hours: 65

Maximum priority (dgi dependencies):
1. #362 Save and Exit icon and REST Endpoints

Regular:
2. #363 Add Playwrights sample to template gallery.
3. #360 Chrome: clicking on a tag in the structure panel doesn't turn it bold
4. #361 Pasted tags not showing up in the structure panel
5. #264 Pop-up warning message when copying/pasting
6. #171 Add copy tag and contents to the right-click options

CWRC:
7. #301 Avoiding tag redundancy
8. #343 rationalize top menu (input from CWRC pending)

**Comment by jchartrand** [comment]
Thanks Mihaela,

Not sure what should be done with the max priority one, though.  Andrew added the button a week ago, and asked (in the GitHub issue) if anything further was needed.  Are we waiting on DGI for more info?

james

**Comment by ilovan** [comment]
Hi James, 
For now, we're waiting for confirmation from dgi that this satisfies the prerequisite. If there's anything else they need from OpenSky, I will ad it to #362 . I left it at the top right now because if anything else is required, it should still be at the top of the priorities list. 
Thanks

**Comment by ilovan** [comment]
Wed Nov 18, 2015

Maximum priority:
1. #328 Add option to settings panel: "Show tags and attributes".
2. #293 Push DEV to TEST

**Comment by ilovan** [comment]
Monday, April 17, 2016:

Maximum priority:
1. #364 (insert tags options)
2. #365 (mini–CWRC-Writer)

**Comment by ilovan** [comment]
Thursday, April 21, 2016:

Maximum priority:
1. #368 Entity URIs - review code and behaviour
2. #366 URLs to test note behaviour in read-only and annotation-only modes

Hold until I had a chance to review the note behaviour in read-only and annotation-only modes:
3. #367 Send pull request to dgi for issues #365  and #364 

Completed from last set of priorities (thank you):
1. #364 (insert tags options)
2. #365 (mini–CWRC-Writer)

**Comment by ilovan** [comment]
Monday, May 16, 2016:

1. #369 Hyperlink of link annotations
2. #384 Allow selecting multiple tags in the structure panel 
3. #378 Insert tag outside
4. #183 Add option to split current tag where the cursor is positioned
5. #383 Merge two identical consecutive tags
6. #386 Read external schemas from any server


7. #376 Entity tag equivalency for TEI-L (CWRC dependency)
8. #381 Validator down for standalone CWRC_Writer (CWRC dependency)

**Comment by SusanBrown** [comment]
Hi Andrew and James, 
Sorry there has been a lag in providing the list of priorities. We will try to get back into the swing of that. Here are a few to get going on. Looks like we need to do some testing of ones that Andrew has worked on.

#328 Add option to settings panel: "Show tags and attributes" now relabelled to Revise Selection Panel
#407 In the embedded CWRC-Writer, the notification of what mode you are in pushes the window down high priority interface
#411 Validation popup in standalone
#410 Validation popup in embedded version
#412  Edit source appearing below (this problem seems to be general to many popups and may require help from Digital Echidna but if you can address it that would be great.
#399 Insertion of new paragraph now shown
#403 Error recovery (if possible)  (This is related to #402)
#398 Crumbtrail to select tag
#389 Place entity handling -- there are a bunch of these that we need to address ( #388, #387 too) . Can/should they be addressed one by one or is there something more general that would be a group fix?

**Comment by SusanBrown** [comment]
Priority for Andrew: 
#414 -- answer to Jeff about repo
#438 CSS for notes -- suggestion of how to do this, then over to Abi to implement the change
#369 hyperlinking of link annotations
[likely related to #375 entity panel link to source: works but needs REF in right place for VIAF entities]
#440 CSS chronstructs
#379 automatic scroll
#378 insert tag outside
#441 make context menu order and look consistent

Digital Echidna
#407 In the embedded CWRC-Writer, the notification of what mode you are in pushes the window down
#412 Edit source appearing below (this problem seems to be general to many popups and may require help from Digital Echidna but if you can address it that would be great.
#410 Validation popup in embedded version
#413 Full-screen function **[should be added to DE requests]**

Jeff
#414 expansion of headings panel in cwrc-reader

To test 
#399 Insertion of new paragraph not shown 
#424 
#425 
#316 
#399 in embedded version

Assess severity: on hold
#389 Place entity handling -- there are a bunch of these that we need to address ( #388, #387 too) . Can/should they be addressed one by one or is there something more general that would be a group fix?

Queries/Scoping questions:
#384 multiple tag selection/dragging in structure panel (if answer is confirmation of my understanding, it resolves #385 too)

Post-launch:
#403 Error recovery (if possible) (This is related to #402)
#398 Crumbtrail to select tag
#309 Note popup and default mode
#376 entity tag equivalence 
#320 help popup for overlaps (needs input from Susan)
#311 citation functionality
#418 progress bar for loading

**Comment by SusanBrown** [comment]
Priority for Andrew:

bibcits (various issues--need to talk them through)
notes (ditto)
#369 hyperlinking of link annotations[may require meeting re: autoconversion question; see #427 #309 removed popup  for mini-cwrcwriter
#389 Place and bibcit entity handling -- there are a bunch of these that we need to address ( #388, #387 #457). Can/should they be addressed one by one or is there something more general that would be a group fix?
#464 Clicks produce weird scrolling in CWRC-Reader
#460 Search within CWRC-Reader reading panel
#462 schema object association error
#415 have "CWRC-Reader v0.9" in Reading mode, and "CWRC-Writer v0.9" [note increment in version number] in Writing mode
Parallel image and cwrcwriter mode

lower priority
#399 Insertion of new paragraph not shown
#465 show entity labels not strings
#384 multiple tag selection
#385 multiple tag dragging and dropping (evaluate scope first)
#459 odd selection error -- if fairly quickly diagnosed & fixed. If not then we'll defer it. 

Digital Echidna:
#413 Full-screen function [should be added to DE requests]

To test
#316 add tag functionality to entity tags

Post-launch [*still under review by Susan*]
#403 Error recovery (if possible) (This is related to #402)
#398 Crumbtrail to select tag
#309 Note popup and default mode
#376 entity tag equivalence
#320 help popup for overlaps (needs input from Susan)
#311 citation functionality

**Comment by jchartrand** [comment]
March 17, 2017:

#476  
#376

**Comment by SusanBrown** [comment]
Priorities for @ajmacdonald 
#476
#376

for @colinraybrian to test (in order of priority):
#425 Validation error
#420, #454 and #455 highlighting/locating of bibcits 
#464 Clicks in reading pane produce scrolling jumps in CWRC-Reader
#301 tag redundancy in Orlando place tags
#469 Editor panel pop-ups incorrectly positioned when panel is scrolled
#316 add tag functionality to entity tags
#369 hyperlinking (will require adding xref tags to a doc in oxygen and then uploading and opening in CWRC-Writer to see if it converts it to an entity)
#361 Pasted tags not showing up 

Notes to self: 
Went through all high-priority items
Still to process/review: Nov 27th list

#351 pending response

Lower priority: 
#384 #385 (multiple consecutive tags: select and drag)

Susan to review:
#457 bibcits not inserting around selected prose - standalone and embedded version
#456 editing of bibcits
#331 review tag insertion/deletion options


Post-launch [still under review by Susan]
#403 Error recovery (if possible) (This is related to #402)
#398 Crumbtrail to select tag
#309 Note popup and default mode
#376 entity tag equivalence
#320 help popup for overlaps (needs input from Susan)
#311 citation functionality
